src/host-config/types.ts

```diff
@@ -0,0 +1,86 @@
+import * as t from 'io-ts';
+import { DoubleQuotedString, NumericIdentifier, withDefault } from '../types';
+
+/**
+ * The internal object representation of the base {...} block
+ * of redsocks.conf, part of RedsocksConfig.
+ * Mostly nonconfigurable except for log_debug (TODO)
+ */
+export const BaseConfig = t.type({
+	log_debug: t.literal('off'),
+	log_info: t.literal('on'),
+	log: t.literal('stderr'),
+	daemon: t.literal('off'),
+	redirector: t.literal('iptables'),
+});
+export type BaseConfig = t.TypeOf<typeof BaseConfig>;
+
+/**
+ * The internal object representation of the redsocks {...} block
+ * of redsocks.conf, with configurable fields only, part of RedsocksConfig.
+ */
+export const WriteableProxyConfig = t.intersection([
+	t.type({
+		type: t.union([
+			t.literal('socks4'),
+			t.literal('socks5'),
+			t.literal('http-connect'),
+			t.literal('http-relay'),
+		]),
+		ip: t.string,
+		port: NumericIdentifier,
+	}),
+	t.partial({
+		login: DoubleQuotedString,
```

Comment:
Why do we need to validate that the strings are double quoted? When coming from the user, they are plain strings. When reading from redsocks.conf, we don't really need to validate them, we just need to remove the quotes if they exist, right?